android/app/src/main/assets/Resources/app.js

```diff
@@ -1,60 +1,48 @@
 'use strict';
 
-// this sets the background color of the master UIView (when there are no windows/tab groups on it)
-Titanium.UI.setBackgroundColor('#000');
-
-// create tab group
-var tabGroup = Titanium.UI.createTabGroup();
-
-//
-// create base UI tab and root window
-//
-var win1 = Titanium.UI.createWindow({
-	title: 'Tab 1',
-	backgroundColor: '#fff'
-});
-var tab1 = Titanium.UI.createTab({
-	icon: 'KS_nav_views.png',
-	title: 'Tab 1',
-	window: win1
-});
-
-var label1 = Titanium.UI.createLabel({
-	color: '#999',
-	text: 'I am Window 1',
-	font: { fontSize: 20, fontFamily: 'Helvetica Neue' },
-	textAlign: 'center',
-	width: 'auto'
-});
-
-win1.add(label1);
-
-//
-// create controls tab and root window
-//
-var win2 = Titanium.UI.createWindow({
-	title: 'Tab 2',
-	backgroundColor: '#fff'
-});
-var tab2 = Titanium.UI.createTab({
-	icon: 'KS_nav_ui.png',
-	title: 'Tab 2',
-	window: win2
+function openWindow() {
+	Ti.UI.setBackgroundColor('#000');
+	var tabGroup = Ti.UI.createTabGroup();
+	var win1 = Ti.UI.createWindow({
+		title: 'Tab 1',
+		backgroundColor: '#fff'
+	});
+	var tab1 = Ti.UI.createTab({
+		icon: 'KS_nav_views.png',
+		title: 'Tab 1',
+		window: win1
+	});
+	win1.add(Ti.UI.createLabel({
+		color: '#999',
+		text: 'I am Window 1',
+		font: { fontSize: 20, fontFamily: 'Helvetica Neue' },
+		textAlign: 'center',
+		width: 'auto'
+	}));
+	var win2 = Ti.UI.createWindow({
+		title: 'Tab 2',
+		backgroundColor: '#fff'
+	});
+	var tab2 = Ti.UI.createTab({
+		icon: 'KS_nav_ui.png',
+		title: 'Tab 2',
+		window: win2
+	});
+	win2.add(Ti.UI.createLabel({
+		color: '#999',
+		text: 'I am Window 2',
+		font: { fontSize: 20, fontFamily: 'Helvetica Neue' },
+		textAlign: 'center',
+		width: 'auto'
+	}));
+	tabGroup.addTab(tab1);
+	tabGroup.addTab(tab2);
+	tabGroup.open();
+}
+
+Ti.UI.addEventListener('sessionbegin', function () {
```

Comment:
Note sure how to avoid this, but it feels repetitive to hang an event listener and check a property to open the UI.

Reply:
So, instead of doing this...
```javascript
//app.js
Ti.UI.addEventListener('sessionbegin', function () {
	openWindow();
}
if (Ti.UI.hasSession) {
	openWindow();
}
```

We could do the below instead and have our "ti.main.js" always fire a "sessionbegin" event after requiring-in the "app.js"
```javascript
//app.js
Ti.UI.addEventListener('sessionbegin', function () {
	openWindow();
}
```
```javascript
//ti.main.js
require("app.js");
if (Ti.UI.hasSession) {
	Ti.UI.fireEvent('sessionbegin');
}
```

However, I don't really like this. I only wanted the "sessionbegin" event to be fired when our new "tiapp.xml" property "run-in-background" is `true`. And when "run-in-background" is `false`, we only start-up the JS runtime **after** the UI session begins natively, meaning this event isn't relevant in this mode... unless... we want to support multiple UI sessions in parallel in the future.

Having he `Ti.UI.hasSession` if-check in the "app.js" (the way I currently have it) means we can easily enable/disable the "run-in-background" property without having to change the "app.js" code. It's compatible with both modes, which means we can stick to 1 app code template for both.